index.bs

```diff
@@ -316,57 +316,61 @@ Interface {#interface}
 =================
 
 <pre class="idl">
-dictionary NavigatorUABrandVersionDict {
-  required DOMString brand;
+dictionary NavigatorUABrandVersion {
+  DOMString brand;
   DOMString version;
 };
 
+dictionary UADataOptions {
+  boolean platform = false;
+  boolean platformVersion = false;
+  boolean architecture = false;
+  boolean model = false;
+};
+
+dictionary UADataValues {
+  DOMString platform = "";
+  DOMString platformVersion = "";
+  DOMString architecture = "";
+  DOMString model = "";
+};
+
 [Exposed=Window]
 interface NavigatorUAData {
-  readonly attribute FrozenArray&lt;NavigatorUABrandVersionDict&gt; brand;
+  readonly attribute FrozenArray&lt;NavigatorUABrandVersion&gt; UASet;
   readonly attribute boolean mobile;
-  Promise&lt;DOMString&gt; getPlatform();
-  Promise&lt;SOMString&gt; getPlatformVersion();
-  Promise&lt;DOMString&gt; getArchitecture();
-  Promise&lt;DOMString&gt; getModel();
+  Promise&lt;UADataValues&gt; getHighEntropyValues(optional UADataOptions options);
 };
 
 interface mixin NavigatorUA {
-  [SecureContext] NavigatorUAData getUserAgent();
+  [SecureContext] readonly attribute NavigatorUAData userAgentData;
 };
 Navigator includes NavigatorUA;
 
 </pre>
 
 Processing model {#processing}
 --------------
-<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method MUST run these steps:
+On getting, the {{NavigatorUAData/UASet}} attribute must return the set of {{NavigatorUABrandVersion}} objects, initialized with the user agent's [=user agent/brand=] and [=user agent/significant version=].
```

Comment:
This doesn't describe how to go from https://wicg.github.io/ua-client-hints/#user-agent-brand and https://wicg.github.io/ua-client-hints/#user-agent-significant-version to a set. In particular both this and getHighEntryValues() use the same wording.

Reply:
yeah. Added a more explicit algorithm.